drivers/can/can_sam0.c

```diff
@@ -0,0 +1,188 @@
+	.get_max_filters = can_mcan_get_max_filters,
+	.get_max_bitrate = can_mcan_get_max_bitrate,
+	.set_state_change_callback =  can_mcan_set_state_change_callback,
+	.timing_min = {
+		.sjw = 0x1,
+		.prop_seg = 0x00,
+		.phase_seg1 = 0x01,
+		.phase_seg2 = 0x01,
+		.prescaler = 0x01
+		},
+	.timing_max = {
+		.sjw = 0x7f,
+		.prop_seg = 0x00,
+		.phase_seg1 = 0x100,
+		.phase_seg2 = 0x80,
+		.prescaler = 0x200
+		},
+#ifdef CONFIG_CAN_FD_MODE
+	.set_timing_data = can_mcan_set_timing_data,
+	.timing_data_min = {
+		.sjw = 0x01,
+		.prop_seg = 0x00,
+		.phase_seg1 = 0x01,
+		.phase_seg2 = 0x01,
+		.prescaler = 0x01
+		},
+	.timing_data_max = {
+		.sjw = 0x10,
+		.prop_seg = 0x00,
+		.phase_seg1 = 0x20,
+		.phase_seg2 = 0x10,
+		.prescaler = 0x20
+		}
+#endif /* CONFIG_CAN_FD_MODE */
+};
+
+#define SAM0_CAN_IRQ_CONNECT(inst, m)					\
+	do {								\
+		IRQ_CONNECT(DT_INST_IRQ_BY_IDX(inst, m, irq),		\
+			    DT_INST_IRQ_BY_IDX(inst, m, priority),		\
+			    can_sam0_line_x_isr,				\
+			    DEVICE_DT_INST_GET(inst), 0);			\
+		irq_enable(DT_INST_IRQ_BY_IDX(inst, m, irq));		\
+	} while (false)
+
+#define CAN_SAM0_IRQ_CFG_FUNCTION(inst)                                                         \
+static void config_can_##inst##_irq(void)                                                      \
+{                                                                                              \
+	SAM0_CAN_IRQ_CONNECT(inst, 0);                                            \
+}
+
+#define CAN_SAM0_CFG_INST(inst)						\
+	static const struct can_sam0_config can_sam0_cfg_##inst = {		\
+		.mclk = (volatile uint32_t *)MCLK_MASK_DT_INT_REG_ADDR(inst),	\
+		.mclk_mask = BIT(DT_INST_CLOCKS_CELL_BY_NAME(inst, mclk, bit)),	\
+		.gclk_core_id = DT_INST_CLOCKS_CELL_BY_NAME(inst, gclk, periph_ch),\
+		.divider = DT_INST_PROP(inst, divider),			\
+		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(inst),		\
+		.config_irq = config_can_##inst##_irq,		\
+	};						\
+									\
+	static const struct can_mcan_config can_mcan_cfg_##inst =	\
+		CAN_MCAN_DT_CONFIG_INST_GET(inst, &can_sam0_cfg_##inst);
+
+#define CAN_SAM0_DATA_INST(inst)						\
+	static struct can_sam0_data can_sam0_data_##inst;			\
+									\
+	static struct can_mcan_data can_mcan_data_##inst =		\
+		CAN_MCAN_DATA_INITIALIZER(&can_sam0_data_##inst.msg_ram, \
+					  &can_sam0_data_##inst);	\
+
+#define CAN_SAM0_DEVICE_INST(inst)					\
+	DEVICE_DT_INST_DEFINE(inst, &can_sam0_init, NULL,		\
+			      &can_mcan_data_##inst,			\
+			      &can_mcan_cfg_##inst,			\
+			      POST_KERNEL, CONFIG_CAN_INIT_PRIORITY,	\
+			      &can_sam0_driver_api);
+
+#define CAN_SAM0_INST(inst)                                                     \
+	PINCTRL_DT_INST_DEFINE(inst);                                          \
+	CAN_SAM0_IRQ_CFG_FUNCTION(inst)                                         \
+	CAN_SAM0_CFG_INST(inst)                                                 \
+	CAN_SAM0_DATA_INST(inst)                                                \
```

Comment:
Keep `\` vertically align.
Use only TABs for this purpose
Loop longest line and add 1 TAB+\
Then repeat for the others.
Zephyr is 8 space TAB like Linux.

```diff
#define SAM0_CAN_IRQ_CONNECT(inst, m)							\
...
+		.gclk_core_id = DT_INST_CLOCKS_CELL_BY_NAME(inst, gclk, periph_ch),	\
		.divider = DT_INST_PROP(inst, divider),					\
		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(inst),				\
		.config_irq = config_can_##inst##_irq,					\
	};										\
...
#define CAN_SAM0_DEVICE_INST(inst)							\
	DEVICE_DT_INST_DEFINE(inst, &can_sam0_init, NULL,				\
			      &can_mcan_data_##inst,					\
			      &can_mcan_cfg_##inst,					\
			      POST_KERNEL, CONFIG_CAN_INIT_PRIORITY,			\
			      &can_sam0_driver_api);
...
#define CAN_SAM0_INST(inst)								\
	PINCTRL_DT_INST_DEFINE(inst);							\
	CAN_SAM0_IRQ_CFG_FUNCTION(inst)							\
	CAN_SAM0_CFG_INST(inst)								\
	CAN_SAM0_DATA_INST(inst)							\
```